src/cleanlab_tlm/utils/chat.py

```diff
@@ -141,7 +187,7 @@ def _form_prompt_responses_api(messages: list[dict[str, Any]], tools: Optional[l
                 output += f"<tool_response>\n{json.dumps(tool_response, indent=2)}\n</tool_response>\n\n"
         else:
             role = msg.get("name", msg["role"])
```

Comment:
this logic is duplicated in two places, you should define a `_get_prefix()` helper method and use it throughout:
Something like:

```
output += _get_prefix(msg)
```